[Vignesh Sankar] iP

vigneshsankariyer1234567890 opened the pull request:
## DukePro
> “Your mind is for having ideas, not holding them.” 
> – David Allen (source)

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,
1. download it from here.
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

It is also absolutely **FREEEEEEEEE!!!**

Features include 
- [X] Todos
- [X] Events
- [X] Deadlines
- [X] Saving your various tasks
- [X] GUI
- [ ] Updating Tasks (Coming soon!)

You can use the Duke project to practice your ~~Javascript~~ Java programming.

```java
public class Main {
    public static void main(String[] args) {
        System.out.println("Hello World!");
    }
}
```

Commit:
Add Gradle support

vigneshsankariyer1234567890 committed:
Add Level-1

vigneshsankariyer1234567890 committed:
Add Level-2

vigneshsankariyer1234567890 committed:
Add A-Classes

Introduced a new class, Task

vigneshsankariyer1234567890 committed:
Add Level-3

vigneshsankariyer1234567890 committed:
Add Level-4

Made Task abstract and added subclasses to Task.

This allows for greater readability and usage of SE Principles.

Made use of Enums to define Task Type.

This allows for greater readability and usage of SE Principles.

vigneshsankariyer1234567890 committed:
Add A-TextUiTesting

vigneshsankariyer1234567890 committed:
Add Level-5

Added support for incorrect input for deadline and event as well. Removed smiley face.

vigneshsankariyer1234567890 committed:
Add Level-6

vigneshsankariyer1234567890 committed:
Update ACTUAL.txt

vigneshsankariyer1234567890 committed:
Remove id and let id be controlled by Duke

vigneshsankariyer1234567890 committed:
Refactor list of Tasks to TaskList

vigneshsankariyer1234567890 committed:
Add Level-7 Save

vigneshsankariyer1234567890 committed:
Update TaskList.java

vigneshsankariyer1234567890 committed:
Add Level-8: Dates and Times

Dates and times were originally stored as a string. This allows for greater control over inputs into Duke.

vigneshsankariyer1234567890 committed:
Merge branch 'branch-Level-8'

vigneshsankariyer1234567890 committed:
Add Super Class to group Deadline and Event

vigneshsankariyer1234567890 committed:
Add A-MoreOOP

Refactor response method to Parser object and add inheritance from Command

vigneshsankariyer1234567890 committed:
Add A-Packages

Refactor classes into packages.

vigneshsankariyer1234567890 committed:
Add ability to change location of data text

vigneshsankariyer1234567890 committed:
Add A-JUnit testing on task creation

vigneshsankariyer1234567890 committed:
Add A-Jar file

vigneshsankariyer1234567890 committed:
Add JavaDocs for command package

vigneshsankariyer1234567890 committed:
Add JavaDocs for task package

vigneshsankariyer1234567890 committed:
Add JavaDoc for classes

vigneshsankariyer1234567890 committed:
Add A-CodingStandard

vigneshsankariyer1234567890 committed:
Add Level-9 Find

vigneshsankariyer1234567890 committed:
Merge branch 'branch-A-CodingStandard'

vigneshsankariyer1234567890 committed:
Merge branch 'branch-Level-9'

vigneshsankariyer1234567890 committed:
Change name of list of tasks to tasks

vigneshsankariyer1234567890 committed:
Merge branch 'add-gradle-support'

vigneshsankariyer1234567890 committed:
Change build.gradle to use original package

vigneshsankariyer1234567890 committed:
Add checkstyle and suppressions

vigneshsankariyer1234567890 committed:
Update build.gradle

Add javafx dependencies

timothywongej commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,86 @@
+    }
+
+    /**
+     * The method that runs the process until a "bye" command is input.
```
Method header comments should start in the form ```Returns ...```, ```Sends ...```, ```Runs ...```

timothywongej commented on src/main/java/duke/Formatter.java:
```diff
@@ -0,0 +1,50 @@
+public class Formatter {
+
+    /**
+     * Method that formats a String to a Task object.
```
Method header comment should start with ```Formats```

timothywongej commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,80 @@
+
+    public TaskList filter(String input) {
+        TaskList taskList = new TaskList();
+        for ( int i = 0; i < this.size() ; i++ ) {
```
Remove white space

timothywongej commented on src/main/java/duke/UserInterface.java:
```diff
@@ -0,0 +1,45 @@
+    Scanner sc;
+
+    UserInterface() {
+
```
Remove line (several other methods in this class as well)

timothywongej reviewed (commented):
Looks good. 👍
Overall, I found the OO logic of your code impressive. I noted a few minor coding standard violations. Most method header comments have to be changed to start in the form of ```Returns``` (not ```Return...``` or ```Method returns...```).

yuifuku1118 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,70 @@
+                    throw new DukeException("You need to include a search word.");
+                }
+                if (words.length > 2) {
+                    throw new DukeException("You can only include 1 search word.");
```
Maybe you could use else for simplicity?

yuifuku1118 commented on src/main/java/duke/UserInterface.java:
```diff
@@ -0,0 +1,45 @@
+        return sc.nextLine();
+    }
+
+    public void printInitialGreeting() {
+        printLogo();
+        System.out.println("Hello I'm Duke\n" + "What can I do for you?");
+    }
+
+    public void printGoodByeGreeting() {
+
+        System.out.println("Bye. Hope to see you again soon!");
+    }
+
+    public void showError(DukeException e) {
+
+        System.out.println("OOPS!!! " + e.getMessage());
+    }
+
+    public void showResult(CommandResult commandResult) {
+        System.out.println(commandResult.getFeedbackResult());
+    }
+
+    private void printLogo() {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);
+    }
```
I like how you named your method name with verbs, easy to understand at one glance!

yuifuku1118 reviewed (commented):
LGTM!!! Well organized and well commented!
Just added few comments and possible improvements.

vigneshsankariyer1234567890 committed:
Level-10

vigneshsankariyer1234567890 committed:
Update MainWindow.java to exit

vigneshsankariyer1234567890 committed:
Refactor into packages

There are several classes which are not organised into packages.

Refactoring these classes into different packages improves code structure.

As a step towards such unification, we can organise common classes into packages. Doing so will make it easier for readers to understand the groups of classes within the code.

vigneshsankariyer1234567890 committed:
Improve the styling of the classes

There are several errors with the styling of the classes.

Improving the styling will allow the checkStyle to pass and improve the coding standards.

As a step towards such improvement, let's review the styling of the classes and correct them. This will allow for higher coding standards.

vigneshsankariyer1234567890 committed:
Improve tests for Parser and Storage

Unit testing requirements for Parser and Storage are not fulfilled yet. Developer Unit Testing should be done in order verify the function of the Parser and Storage classes.

Improving tests for Parser and Storage will confirm the functionality of both classes and reduce chances of regression.

As a step towards test-driven development, let's build tests in order to test and verify that these classes work.

vigneshsankariyer1234567890 committed:
Use assertions to verify functions

vigneshsankariyer1234567890 committed:
Merge pull request #1 from vigneshsankariyer1234567890/branch-A-Assertions

Use assertions to verify functions

vigneshsankariyer1234567890 committed:
Refactor Classes to improve code quality

vigneshsankariyer1234567890 committed:
Merge pull request #2 from vigneshsankariyer1234567890/branch-A-CodeQuality

Refactor Classes to improve code quality

vigneshsankariyer1234567890 committed:
Add Updating of Tasks

Adding class to support extension of IP: Update Tasks.

Previously, updating tasks would require the deletion and re-addition of tasks. A new UpdateCommand class should be created to support the new requirement.

Creating and adding a field to the parser will allow for easy updates of Tasks without having to change anything else. This completes the CRUD functionality.

As a step towards this feature, let's create a new Command, UpdateCommand, and add a case to be handled by the parser.

vigneshsankariyer1234567890 committed:
Merge pull request #3 from vigneshsankariyer1234567890/branch-C-UpdateTasks

Add Updating of Tasks

vigneshsankariyer1234567890 committed:
Refactor Methods and Classes for readability

vigneshsankariyer1234567890 committed:
Refactor command logic for parser

vigneshsankariyer1234567890 committed:
Improve class styling

vigneshsankariyer1234567890 committed:
Add Help Command to give more guidance to user

vigneshsankariyer1234567890 committed:
Remove truncation of text for Duke outputs

vigneshsankariyer1234567890 committed:
Add javadoc

vigneshsankariyer1234567890 committed:
Add Ui.png

vigneshsankariyer1234567890 committed:
Add User Guide
